### PR TITLE
python3Packages.scikit-misc: init at 0.1.4

### PR DESCRIPTION
**Pull request by samuela**
###### Motivation for this change
Add scikit-misc python package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137464` at 32d383d7 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.scikit-misc</li>
    <li>python39Packages.scikit-misc</li>
  </ul>
</details>

**Comment by risicle** [comment]
Builds happily on macos 10.15 & nixos x86_64

File: pkgs/development/python-modules/scikit-misc/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  nativeBuildInputs = [ cython gfortran numpy ];
+
+  # Testing packages that involve cython is a little tricky. We need to point
+  # pytest to the output directory instead of the source directory. See
+  #   * https://github.com/NixOS/nixpkgs/pull/137038
+  #   * https://discourse.nixos.org/t/cant-import-cythonized-modules-at-checkphase/14207
+  # for more information.
+  checkInputs = [ pytest-cov pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/scikit-misc/default.nix:30]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/scikit-misc/default.nix (hunk 1 above)
Comment (samuela) [reply pkgs/development/python-modules/scikit-misc/default.nix]: Done in https://github.com/NixOS/nixpkgs/pull/137464/commits/2a9a4a57c5b5a090bc72809d1db482c227498e31.

File: pkgs/development/python-modules/scikit-misc/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1h8lxyazvq3b2c8blj131zn8wkbzjb8ynpr2a7yaanblfsbzdv92";
+  };
+
+  postPatch = ''
+    substituteInPlace pytest.ini --replace "--cov --cov-report=xml" ""
+  '';
+
+  nativeBuildInputs = [ cython gfortran numpy ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/scikit-misc/default.nix:26]: numpy is a runtime dependency https://github.com/has2k1/scikit-misc/search?q=numpy

Comment (samuela) [reply pkgs/development/python-modules/scikit-misc/default.nix:26]: Just to clarify: would you prefer to see numpy removed from `nativeBuildInputs` and put in `propagatedBuildInputs` instead?

**Comment by NickCao** [comment]
already in nixpkgs.

**Pull request closed without merging**